### PR TITLE
Add pre commit hooks for `lint` and `fix`

### PR DESCRIPTION
**Pull request by dmateusp**
Test it out :)

```yaml
repos:
- repo: https://github.com/dmateusp/sqlfluff
  rev: 9cb4a120ed877a3abdfbef054d00763997e73069
  hooks:
    - id: sqlfluff-lint
      # For dbt projects, this installs the dbt "extras"
      # additional_dependencies: ['.[dbt]']
    - id: sqlfluff-fix
      args: [--rules, "L003,L014"]
      # additional_dependencies: ['.[dbt]']
```

Related: https://github.com/sqlfluff/sqlfluff/issues/523

### CODE REVIEW

File: docs/source/production.rst
Comment (NiallRees) [inline docs/source/production.rst:100]: Maybe a note on where to put this file?

File: docs/source/production.rst
Comment (NiallRees) [inline docs/source/production.rst:113]: A note to explain these are arbitrary and that you can configure any rules you like might be helpful

File: docs/source/production.rst
Comment (NiallRees) [inline docs/source/production.rst:109]: Explain when you need this or not? Or link to somewhere else in the docs that does

File: docs/source/production.rst
Comment (alanmcruickshank) [inline docs/source/production.rst:98]: Great feature, however I think this needs to be fleshed out a little. My assumption is that a lot of sql users (and therefore sqlfluff users) won't know what a `pre-commit` hook is and when to use one. Either we should flesh out a little more detail here on when you might  want to use this feature or we should link to some external literature on that topic, just so people can make informed decisions about when and how to use these.

**Review by alanmcruickshank — CHANGES_REQUESTED** [review]
Wow - surprised that this was so easy. Good work 🙌 .

I think the docs need fleshing out a bit more so that people can make informed decisions, but the code is sound.

File: .pre-commit-hooks.yaml
Comment (pwildenhain) [inline .pre-commit-hooks.yaml:13]: In the latest version, can we just say `sqlfluff fix` and it will automatically apply all rules? I didn't know we had implemented that yet

Comment (alanmcruickshank) [reply .pre-commit-hooks.yaml:13]: With #583 I think this ambiguity has been resolved

Comment (dmateusp) [reply .pre-commit-hooks.yaml:14]: I added a comment to mention that all the rules are applied by default

File: docs/source/conf.py
Comment (dmateusp) [inline docs/source/conf.py:33]: Found that we were behind on versions in the docs, I added this "stable_version" @alanmcruickshank.

I'm using that in the "example" of a pre-commit configuration. Didn't want maintainers to have to think about bumping the version in multiple places

File: docs/source/production.rst
Comment (dmateusp) [reply docs/source/production.rst:120]: Added a note https://github.com/sqlfluff/sqlfluff/pull/576/files#diff-4fc96f7f0c2ce176c6648464700d7924432d594701d88e49facbe97ec6d45fd9R127

File: docs/source/conf.py
Comment (NiallRees) [reply docs/source/conf.py:33]: I think that merging this will result in the live docs stable version becoming an alias for 0.4.0 which isn't released yet. I could be wrong though - @alanmcruickshank ?

Comment (dmateusp) [reply docs/source/conf.py:33]: No, you're right! I should change that back for now

File: src/sqlfluff/config.ini
Comment (dmateusp) [inline src/sqlfluff/config.ini:4]: @NiallRees : changed now :)

**Review by NiallRees — APPROVED** [review]

File: docs/source/conf.py
Comment (barrywhart) [reply docs/source/conf.py:33]: @dmateusp: Did you mean to change this back before merging? It looks like it still says `stable_version` rather than `0.3.6`.

Comment (NiallRees) [reply docs/source/conf.py:33]: @barrywhart https://github.com/sqlfluff/sqlfluff/pull/576/files#diff-3a64d1a0f41d37628ed69b17636e138abf6b56b19175f9d80cce28df4ccbbd42R4

File: docs/source/production.rst
Comment (barrywhart) [inline docs/source/production.rst:117]: IIUC, the `v` should be removed. It looks like `pre-commit` attempts to check out a tag or branch with this name, but SQLFluff branches thus far do not use a `v` prefix, e.g.:

```
git tag --list
0.0.1
0.0.2
0.0.3
0.0.6
0.0.7
0.1.0
0.1.4
0.1.5
0.2.0
0.2.1
0.2.2
0.2.3
0.2.4
0.3.0
0.3.1
0.3.2
0.3.3
0.3.4
0.3.5
0.3.6
```

Comment (barrywhart) [reply docs/source/production.rst:117]: This worked for me:
```
-   repo: https://github.com/sqlfluff/sqlfluff
    rev: master
    hooks:
    -   id: sqlfluff-lint
    -   id: sqlfluff-fix
```

Of course, once there's a release with the pre-commit hooks, `rev:` should point to that, not `master`. I was just trying to get this working for one of our projects to try it out. This is great stuff! I've been wanting pre-commit support for a long time.

File: docs/source/conf.py
Comment (barrywhart) [reply docs/source/conf.py:33]: Got it, thanks!

File: docs/source/production.rst
Comment (dmateusp) [reply docs/source/production.rst:117]: @barrywhart thanks for pointing out the problem with `v` in the docs, the pre-commit I had set up with the "mirror" added `v` as a prefix to releases so I made that mistake in the docs.
https://github.com/sqlfluff/sqlfluff/pull/611

**Pull request merged**
